pystac_client/stac_api_io.py

```diff
@@ -34,6 +34,7 @@ def __init__(
         headers: Optional[Dict[str, str]] = None,
         conformance: Optional[List[str]] = None,
         parameters: Optional[Dict[str, Any]] = None,
+        modifier: Optional[Callable[[Any], Any]] = None,
```

Comment:
If the modifier should receive and return `request.Request`s, shouldn't we call that out in the type signature?